src/components/AddItem.jsx

```diff
@@ -1,27 +1,58 @@
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 import { collection, addDoc } from 'firebase/firestore';
 import { db } from '../lib/firebase';
 import { getUser } from '../storage-utils/storage-utils';
+import { onSnapshot } from 'firebase/firestore';
 
 export default function AddItem() {
   const [userToken] = useState(getUser());
   const [itemName, setItemName] = useState('');
   const [frequency, setFrequency] = useState(7);
+  const [message, setMessage] = useState('');
+  const [docs, setDocs] = useState([]);
+
+  useEffect(() => {
+    const unsubscribe = onSnapshot(collection(db, userToken), (snapshot) => {
+      let snapshotDocs = [];
+      snapshot.forEach((doc) => snapshotDocs.push(doc));
+      setDocs(snapshotDocs);
+    });
+    return () => {
+      unsubscribe();
+    };
+  }, [userToken]);
 
   const submitHandler = async (e) => {
     e.preventDefault();
 
+    const cleanList = docs.map((listItem) =>
+      listItem
+        .data()
+        .item.replace(/[^a-zA-Z\s]/g, '')
+        .replace(/\s+/g, ' ')
+        .toLowerCase(),
+    );
+
+    const cleanItemName = itemName
+      .replace(/[^a-zA-Z\s]/g, '')
```

Comment:
optional suggestion/minor: refactor this to a helper method, as I notice the same "clean" chain applied in two spots

Reply:
Thanks! That's a great call-out. I've refactored this to help "DRY" up the code a bit.